src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java

```diff
@@ -101,60 +117,69 @@ protected void update( ModifiedPomXMLEventReader pom )
+                                + " to " + artifactVersion );
+                    }
+                    getChangeRecorder().recordUpdate( "updateParent", getProject().getParent().getGroupId(),
+                            getProject().getParent().getArtifactId(), getProject().getParent().getVersion(),
+                            artifactVersion.toString() );
+                }
             }
         }
         catch ( InvalidVersionSpecificationException e )
         {
-            throw new MojoExecutionException( "Invalid version range specification: " + version, e );
-        }
-
-        ArtifactVersion artifactVersion;
-        try
-        {
-            artifactVersion = findLatestVersion( artifact, versionRange, null, false, allowDowngrade );
+            throw new MojoExecutionException( "Invalid version range specification: " + initialVersion, e );
         }
         catch ( ArtifactMetadataRetrievalException e )
         {
             throw new MojoExecutionException( e.getMessage(), e );
         }
+    }
+
+    private ArtifactVersion resolveTargetVersion( String initialVersion )
+            throws MojoExecutionException, ArtifactMetadataRetrievalException, InvalidVersionSpecificationException
+    {
+        Artifact artifact = getHelper().createDependencyArtifact( DependencyBuilder
+                .newBuilder()
+                .withGroupId( getProject().getParent().getGroupId() )
+                .withArtifactId( getProject().getParent().getArtifactId() )
+                .withVersion( initialVersion )
+                .withType( "pom" )
+                .build() );
 
-        if ( !shouldApplyUpdate( artifact, currentVersion, artifactVersion, forceUpdate ) )
+        VersionRange targetVersionRange = VersionRange.createFromVersionSpec( initialVersion );
+        if ( targetVersionRange.getRecommendedVersion() != null )
         {
-            return;
+            targetVersionRange = targetVersionRange.restrict(
+                    VersionRange.createFromVersionSpec( "[" + targetVersionRange.getRecommendedVersion() + ",)" ) );
         }
 
-        getLog().info( "Updating parent from " + currentVersion + " to " + artifactVersion.toString() );
-
-        if ( PomHelper.setProjectParentVersion( pom, artifactVersion.toString() ) )
+        ArtifactVersion artifactVersion = findLatestVersion( artifact, targetVersionRange, null,
+                false, allowDowngrade );
+        if ( !shouldApplyUpdate( artifact, getProject().getParent().getVersion(), artifactVersion, forceUpdate ) )
         {
-            getLog().debug( "Made an update from " + currentVersion + " to " + artifactVersion );
-
-            this.getChangeRecorder().recordUpdate( "updateParent", artifact.getGroupId(), artifact.getArtifactId(),
-                                                   currentVersion, artifactVersion.toString() );
+            return null;
```

Comment:
Do we have somewhere a log (can be in debug level) why update is not needed? 
It can be useful for debugging.

Reply:
Actually, yes -- `shouldApplyUpdate` seems to be very chatty. :)